lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,267 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+
+    auto weightTy = weight.getType().cast<RankedTensorType>();
+    if (weightTy.getRank() != 2)
+      return rewriter.notifyMatchFailure(op, "weight must be rank 2");
+
+    auto indicesTy = indices.getType().cast<RankedTensorType>();
+    if (indicesTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "indices must be a vector");
+
+    auto offsetsTy = offsets.getType().cast<RankedTensorType>();
+    if (offsetsTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "offsets much be a vector");
+
+    Type weightElemTy = weightTy.getElementType();
+
+    if(!isSparse){
+      if (modeInt == 0) {
+
+        int64_t iterationMapDimension = weightTy.getRank() + indicesTy.getRank();
+
+        SmallVector<AffineExpr> indicesExpr;
+        indicesExpr.push_back(mlir::getAffineDimExpr(1, context));
+
+        auto indicesIndexingMap =
+            AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                          indicesExpr, context);
+
+        SmallVector<AffineExpr> offsetsExpr;
+        offsetsExpr.push_back(mlir::getAffineDimExpr(0, context));
+
+        auto offsetIndexingMap =
+            AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                          offsetsExpr, context);
+
+        SmallVector<AffineExpr> outputExpr;
+        outputExpr.push_back(mlir::getAffineDimExpr(0, context));
+        outputExpr.push_back(mlir::getAffineDimExpr(2, context));
+
+        auto outputIndexingMap =
+            AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                          outputExpr, context);
+
+        SmallVector<AffineMap, 3> indexingMaps = {
+            indicesIndexingMap,
+            offsetIndexingMap,
+            outputIndexingMap,
+        };
+
+        SmallVector<StringRef> iteratorTypes(iterationMapDimension,
+                                            getParallelIteratorTypeName());
+
+        SmallVector<Value> sizes{getDimOp(rewriter, loc, offsets, 0),
+                                embeddingDim};
+
+        Value initTensor =
+            createZeroInitTensor(rewriter, loc, sizes, weightElemTy);
+
+        Value offsetsLength = getDimOp(rewriter, loc, offsets, 0);
+        Value indicesLength = getDimOp(rewriter, loc, indices, 0);
+
+        Value embeddingBagResult =
+            rewriter
+                .create<linalg::GenericOp>(
+                    loc, initTensor.getType(), ValueRange{indices, offsets},
+                    initTensor,
+                    /*indexingMaps=*/indexingMaps,
+                    /*iteratorTypes=*/iteratorTypes,
+                    [&](OpBuilder &b, Location loc, ValueRange args) {
+                      Value indexInIndices = args[0];
+                      Value offsetsI = args[1];
+                      Value initTensorElem = args[2];
+
+                      Value indexI =
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/0);
```

Comment:
nit: inside the `linalg::GenericOp` you can use `b` instead of `rewriter`. This should make the lines a bit shorter